CHANGES.rst

```diff
@@ -11,6 +11,204 @@ Changelog
+  [thet]
+
+- Add ``safe_encode`` utility function to ``utils`` to safely encode unicode to a specified encoding.
+  The encoding defaults to ``utf-8``.
+  [thet]
+
+- The password reset templates were changed to make use of ``content-core`` macros.
+  [thet]
+
+- Add utility method to retrieve the top most parent request from a sub request.
+  [thet]
+
+- Add ``mockup-patterns-relateditems-upload`` resource, which can be used in custom bundles to add the upload feature in the related items widget.
+  [thet]
+
+- Move ``get_top_site_from_url`` from plone.app.content to ``utils.py`` and make it robust against unicode paths.
+  This function allows in virtual hosting environments to acquire the top most visible portal object to operate on.
+  It is used for example to calculate the correct virtual root objects for Mockup's related items and structure pattern.
+  [thet]
+
+- Add sort_on field to search controlpanel.
+  [rodfersou]
+
+- PLIP 1340: Deprecate portal_quickinstaller.
+  You should no longer use CMFQuickInstallerTool methods, but GenericSetup profiles.
+  See https://github.com/plone/Products.CMFPlone/issues/1340
+  [maurits]
+
+- Include mockup 2.4.0.
+  [thet]
+
+- PasswordResetTool moved from its own package to here (includes cleanup and removal of ``getStats``).
+  [tomgross]
+
+- Prevent workflow menu overflowing in toolbar [MatthewWilkes]
+
+- Add default icon for top-level contentview and contentmenu toolbar entries [alecm]
+
+- Toolbar: Make menu hover background fit whole menu width. [thet]
+
+- Toolbar: Don't force scoll buttons to be left, when toolbar is right. [thet]
+
+- Toolbar: Make first level list items exand the whole toolbar width - also when scroll buttons are shown. [thet]
+
+- Toolbar: Make scroll buttons exand whole toolbar width. [thet]
+
+- Toolbar: Let the toolbar submenus be as wide as they need to be and do not break entries into multiple lines. [thet]
+
+- Resource Registry:
+  In ``debug-mode`` (zope.conf, buildout) do not load cache the production bundle.
+  [jensens]
+
+- Resource Registry:
+  In ``debug-mode`` (zope.conf, buildout) do not ignore development mode for anonymous users.
+  [jensens]
+
+- Resource Registry: If file system version is newer than ``last_compilation`` date of a bundle, use this as ``last_compilation`` date.
+  [jensens]
+
+- New browser view based login code - merged from plone.login (credits to esteele, pbauer, agitator, jensens, et al).
+  `portal_skins/plone_login` is now gone, see PLIP #2092.
+  Also, password reset view moved to login subfolder to keep things together.
+  Some testbrowser based tests needed changes because of z3c.form based login form .
+  The Plone specific, rarely used cross site __ac cookie SSO feature/hack was removed.
+  In case somebody needs this, please make it an addon package.
+  Better use a field proven, more secure way, like OAuth2, Shibboleth or someting similar.
+  [jensens, et al]
```

Comment:
the whole added changes above seems to be odd. w/o looking deeper into it, is it from a rebase , which worked not that well?

Reply:
@jensens I think i missed to cherry-pick your merge-mistake commit from https://github.com/plone/Products.CMFPlone/pull/1963/commits/e261d98f5f95c46c30ed6e296d8f5694dbaedf7c - I will fix it.

The upgrade steps in p.a.upgrade: 
coredev 5.1 and 5.2 run on the same branch of p.a.upgrade (master).
I added steps for v51/to_514 and v52/to_alpha1 as you can see here: https://github.com/plone/plone.app.upgrade/pull/170/files

I tested with data.fs v=5.1.2 on coredev 5.1 (upgrading to 5.1.4)
and coredev 5.2 (upgradeing to latest 5.2)